### ISSUE TITLE
IE7 and IE8 throw an error

### ISSUE THREAD

**Issue by davidgrossi** [kveeiv/extjs-boxselect#11]
Hi !

First of all : great job !
I'm using your component which works fine with ALL browsers except IE7 and IE8. This is a big deal as it's mainly used accross  companies.

This issue has been reported on Moritori's GitHub and you answered to test your last version 2.0 with IE7 / IE8.
That's what I've just done, and unfortunately, those browsers still throw an error.

Regards.

**Comment by kveeiv** [comment]
Can you reproduce this error with 4.1.0 and 2.0.1, and post a test case? I don't have access to IE7 to test with, but I use the examples page as test cases and am able to run those successfully in IE8.

**Comment by davidgrossi** [comment]
Hello again,

This morning, I took the time to reproduce the problem within my application.
I'm using boxselect 2.0.1 and Ext 4.1.1 final.
Works fine with Chrome, Firefox, and IE9.

With IE7 and IE8, the page layout is destroyed (seems serious !) and IE throw an error on the line 12267 of ext-all-debug.js : style[hook.name] = value;

Looks like a styling exception... but I can't find what's wrong exactly.

I've made 3 screenshots to show you how the problem looks, but I don't know where to post them with Github.
Do you have an email so I can send you those screens.

**Comment by davidgrossi** [comment]
Hi Kevin,

Some more debugging shows that in the line style[hook.name] = value, it tries to set a "width" properties to the value "-16px".

Still don't know why at the moment.

**Comment by kveeiv** [comment]
If you could upload these to http://imgtmp.com/ and post the links here, that would be great. Please be sure your IE8 page has the appropriate <!DOCTYPE tag at the beginning. Can you also check the included examples/boxselect.html to see if your issue is replicated there? I use that page as a test page and no errors show up for me in IE8.

**Comment by davidgrossi** [comment]
Hi !

Thanks for your reply.
It's a great link to host tmp images.

Here are :
- Screen of IE9 (OK) : http://f.imgtmp.com/fmqz3.jpg
- Screen of IE8 (not OK) : http://f.imgtmp.com/MPZjp.jpg
- Screen of IE7 (not OK) : http://f.imgtmp.com/Enq04.jpg

Yes, the DocType is correctly set.
I'm using <!DOCTYPE html> in order to escape from IE quircks mode.

I'll try your example as soon as I can, maybe not today because I'm very busy these days.

Best regards,
David Grossi.

**Comment by kveeiv** [comment]
The IE8 and IE7 screenshots have other elements that are out of place. If you comment out the boxselect portion, do they layout properly?

After testing that, can you test a modification to BoxSelect? I put out BoxSelect 2.0.2 yesterday, if you can change line 1627:

```
owner.inputElCt.setStyle('width', inputWidth + 'px');
```

To this:

```
if (inputWidth > 0) {
  owner.inputElCt.setStyle('width', inputWidth + 'px');
}
```

and let me know the results, that might at least narrow down a workaround to avoid the script errors, though the layout will probably still be off.

**Comment by davidgrossi** [comment]
Exact : other elements are disturbed in their layout.
If I comment the code where the boxselect is created, the layout is ok.
Everything works also well if I simply replace "boxselect" with "combobox" xtype.

I'll try your tip within 20 minutes and come back.

**Comment by davidgrossi** [comment]
Great ! Everything is fixed.
Your trick is working fine !

Now, here is the status with Ext JS 4.1.1 :
- IE9 : OK
- IE8 : OK - 100% the same except rounder border of items (I suppose the corresponding CSS is not supported)
- IE7 : OK but a small artifact with a scrollbar appearing even if not necessary : http://f.imgtmp.com/DSg15.jpg

This is REALLY a good job you have done with this widget.
Very clean code, also.
Thanks a lot !

**Comment by kveeiv** [comment]
Could you help me test another possible solution? There was a report on the forum threads about another width issue, and I am curious as to whether or not it will also resolve the IE issue. As I don't have a test case to replicate this, it is difficult for me to test. The change to test would be to replace the entire layout class at the bottom of BoxSelect.js with the layout class given in this post: http://www.sencha.com/forum/showthread.php?134751-Ext.ux.form.field.BoxSelect-Intuitive-Multi-Select-ComboBox&p=853134&viewfull=1#post853134

I would be very appreciative if you could test this with your case, instead of just the "if" check that I had you add, as it is probably a more comprehensive solution if it works.

**Comment by davidgrossi** [comment]
No problem.

As it's 4:45 in the morning now (after a looong coding session), I'm a bit
tired, and I leave tomorrow until sunday evening...
...I promise I'll do this test when I'll come back home on sunday.

Regards,

David Grossi.

---

GSM : +262 (0)692 229 639
Mail : david.grossi@exauce.com
Web : www.exauce.com

---

On Fri, Jul 13, 2012 at 2:12 AM, Kevin Vaughan <
reply@reply.github.com

**Comment by kveeiv** [comment]
I have released 2.0.3 which includes the previously mentioned fixed, thanks for the report and the help in testing!

**Comment by davidgrossi** [comment]
No : thanks to YOU to maintain such a good component.

Best regards,
David Grossi.
